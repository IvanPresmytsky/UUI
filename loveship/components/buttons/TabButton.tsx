import css from './TabButton.module.scss';
import * as uui from '@epam/uui';
import { withMods } from '@epam/uui-core';

export type TabButtonMods = uui.TabButtonProps & {
    theme?: 'light' | 'dark';
};

function applyTabButtonMods(mods: TabButtonMods & uui.TabButtonProps) {
    return [mods.theme === 'dark' && css.themeDark];
}

export type TabButtonProps = uui.TabButtonProps & TabButtonMods;

export const TabButton = withMods<uui.TabButtonProps, TabButtonMods>(uui.TabButton, applyTabButtonMods);
