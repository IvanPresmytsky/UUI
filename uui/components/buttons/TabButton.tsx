import React from 'react';
import * as UuiComponents from '@epam/uui-components';
import { withMods } from '@epam/uui-core';
import { getIconClass } from './helper';
import { CountIndicator } from '../widgets';
import { systemIcons } from '../../icons/icons';
import css from './TabButton.module.scss';

export interface TabButtonMods {
    /**
     * Defines component size.
     * @default '48'
     */
    size?: '36' | '48' | '60';
    /**
     * Defines is the component showing Notify
     */
    withNotify?: boolean;
}

/** Represents the properties of a TabButton component. */
export type TabButtonProps = TabButtonMods & UuiComponents.ButtonProps;

function applyTabButtonMods(mods: TabButtonProps) {
    return [
        css.root,
        'uui-tab-button',
        css['size-' + (mods.size || '48')],
        mods.withNotify && css.withNotify,
        ...getIconClass(mods),
    ];
}

export const TabButton = withMods<UuiComponents.ButtonProps, TabButtonMods>(
    UuiComponents.Button,
    applyTabButtonMods,
    (props) => ({
        dropdownIcon: systemIcons['36'].foldingArrow,
        clearIcon: systemIcons['36'].clear,
        ...props,
        rawProps: { role: 'tab', ...(props.rawProps as any) },
        countIndicator: (countIndicatorProps) => (
            <CountIndicator
                { ...countIndicatorProps }
                color={ props.isLinkActive ? 'info' : 'neutral' }
                size="18"
            />
        ),
    }),
);
