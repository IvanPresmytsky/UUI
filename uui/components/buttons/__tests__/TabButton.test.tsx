import React from 'react';
import { TabButton } from '../TabButton';
import { renderSnapshotWithContextAsync } from '@epam/uui-test-utils';
import { ReactComponent as calendarIcon } from '../../../icons/calendar-18.svg';

describe('TabButton', () => {
    it('should be rendered correctly', async () => {
        const tree = await renderSnapshotWithContextAsync(<TabButton />);
        expect(tree).toMatchSnapshot();
    });

    it('should be rendered correctly with props', async () => {
        const tree = await renderSnapshotWithContextAsync(
            <TabButton
                onClick={ jest.fn }
                icon={ calendarIcon }
                isDisabled={ false }
                withNotify={ true }
            />,
        );
        expect(tree).toMatchSnapshot();
    });
});
