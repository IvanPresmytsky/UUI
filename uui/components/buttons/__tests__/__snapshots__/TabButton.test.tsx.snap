// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TabButton should be rendered correctly 1`] = `
<span
  className="uui-button-box uui-enabled root uui-tab-button uui-size-48 uui-no-left-icon uui-no-right-icon"
  role="tab"
  tabIndex={-1}
/>
`;

exports[`TabButton should be rendered correctly with props 1`] = `
<button
  aria-disabled={false}
  className="uui-button-box uui-enabled -clickable root uui-tab-button uui-size-48 withNotify uui-has-left-icon uui-no-right-icon"
  onClick={[Function]}
  role="tab"
  tabIndex={0}
  type="button"
>
  <div
    className="container uui-icon uui-enabled"
    style={Object {}}
  >
    <svg
      className=""
    />
  </div>
</button>
`;
